### PR TITLE
OOP School Library : Associations

### PR DESCRIPTION
**Pull request by HtetWaiYan7191**
### This pull request includes the following changes : 

- Add book class
- Update person class
- Add one to many associations for Classroom class And Student class 
- Add rental class to join book and person class in many to many relationships

Thanks for reviewing my code 😁

### CODE REVIEW

File: class/book.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class Book
+  attr_accessor :title, :author, :rental
+
+  def initialize(title, author)
+    @title = title
+    @author = author
+    @rental = []
+  end
```
Comment (BregornOriginal) [inline class/book.rb:9]: - [x] Good work implementing the Book class, kindly add the method to `add_rental` as it's requested in the requirements 💯

File: class/person.rb [hunk 2]
```diff
@@ -12,6 +12,7 @@ def initialize(age = nil, name = 'Unknown', parent_permission: true)
     @name = name
     @parent_permission = parent_permission
     @age = age
+    @rental = []
   end
```
Comment (BregornOriginal) [inline class/person.rb:18]: - [x] As we are having a [many-to-many](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#many-to-many-relationship) relationship with rentals, it's necessary to add a method `add_rental` here too

**Review by BregornOriginal — CHANGES_REQUESTED** [review]
Hi @HtetWaiYan7191  

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

giphy (5)

## Highlights

✅ Linters are passing
✅ Good commit history
✅ Good Pull request description

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by thecodechaser — APPROVED** [review]
## STATUS: APPROVED 🟢

## Hi @HtetWaiYan7191👋, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## To Highlight 🎉
- Good job with associations. ✔️
- Gitflow is followed. :heavy_check_mark:
- Professional README file. :heavy_check_mark:
- No linter errors. :heavy_check_mark:
- PR has a good title and summary. ✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**